src/galaxy/SystemBody.cpp

```diff
@@ -262,6 +262,42 @@ bool SystemBody::IsScoopable() const
 	return false;
 }
 
+// all input units are in km
+double SystemBody::ComputeDensity(const double radius, const double atmosphereHeight, const double h, const double baricStep) const
+{
+        int numSamples = 16;
+        double totalHeight = radius + atmosphereHeight;
+        double minHeight = radius + h;
+        double tmax = sqrt(totalHeight*totalHeight - minHeight*minHeight); // maximum t
+        double tmin = -tmax;
+        double length = tmax - tmin;
+        double step = length / numSamples;
+
+        double density = 0.0;
+        for (int i = 0; i < numSamples; ++i) {
+                double t = tmin + step * (i + 0.5);
+                double h = sqrt(minHeight*minHeight + t*t) - radius;
+                density += step * exp(-h / baricStep);
+        }
+
+        return density;
+}
+
+// all input units are in km
+vector3f SystemBody::GetCoefficients(const double radius, const double atmHeight, const double baricStep) const
```

Comment:
I'd strongly prefer this function be annotated with a description of both the math inside as well as its purpose.